nsxt/resource_nsxt_policy_nat_rule.go

```diff
@@ -394,13 +422,15 @@ func resourceNsxtPolicyNATRuleCreate(d *schema.ResourceData, m interface{}) erro
 
 	log.Printf("[INFO] Creating NAT Rule with ID %s", id)
 
-	err := patchNsxtPolicyNATRule(getSessionContext(d, m), connector, gwID, ruleStruct, isT0)
+	err := patchNsxtPolicyNATRule(getSessionContext(d, m), connector, gwID, ruleStruct, isT0, natType)
 	if err != nil {
 		return handleCreateError("NAT Rule", id, err)
 	}
 
 	d.SetId(id)
 	d.Set("nsx_id", id)
+	// In case nat type was not specified or got overridden by action
+	d.Set("type", natType)
```

Comment:
Is this to restore any change that might have been done in line 311?

Reply:
No, this is create flow - if nat type was left blank by the user, we need to set the `Computed` nat type that was determined in line 372